sql/timetype.go

```diff
@@ -120,17 +120,22 @@ func (t timespanType) ConvertToTimespanImpl(v interface{}) (timespanImpl, error)
 			minutes := absValue / 100
 			seconds := absValue % 100
 			if minutes <= 59 && seconds <= 59 {
+				microseconds := (seconds*microsecondsPerSecond) + (minutes * microsecondsPerMinute)
```

Comment:
time.Duration does all these magnitude conversions for you, should use them unless you have very good reasons not to

https://stackoverflow.com/questions/41503758/conversion-of-time-duration-type-microseconds-value-to-milliseconds

Reply:
I remember running into issues using `time.Duration` before. I think it had to do with interpreting numbers as a string and whatnot, and converting that to a `time.Duration` took more code/effort than the current solution. It's something I could revisit in the future, but probably not worth completely rewriting the file for now.